library/src/handle.cpp

```diff
@@ -1,28 +1,78 @@
-
+/* ************************************************************************
+ * Copyright 2016 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
 #include "definitions.h"
 #include "status.h"
 #include "handle.h"
 #include <hip/hip_runtime_api.h>
+#include <unistd.h>
+#include <pwd.h>
 
 /*******************************************************************************
  * constructor
  ******************************************************************************/
-_rocblas_handle::_rocblas_handle()
+_rocblas_handle::_rocblas_handle() : layer_mode(rocblas_layer_mode_logging)
 {
-
     // default device is active device
     THROW_IF_HIP_ERROR(hipGetDevice(&device));
     THROW_IF_HIP_ERROR(hipGetDeviceProperties(&device_properties, device));
 
     // rocblas by default take the system default stream 0 users cannot create
+
+    // set layer_mode from vaule of environment variable ROCBLAS_LAYER
+    char* str_layer_mode;
+    if((str_layer_mode = getenv("ROCBLAS_LAYER")) == NULL)
+    {
+        layer_mode = rocblas_layer_mode_none;
+    }
+    else
+    {
+        int int_layer_mode = atoi(str_layer_mode);
+
+        layer_mode = (rocblas_layer_mode)int_layer_mode;
+    }
+
+    // open log file
+    if(layer_mode & rocblas_layer_mode_logging)
+    {
+        // construct filepath for log file in home directory
+        const char* homedir_char;
+        std::string homedir_str;
+
+        if((homedir_char = getenv("HOME")) == NULL)
+        {
+            homedir_char = getpwuid(getuid())->pw_dir;
+        }
+        if(homedir_char == NULL)
+        {
+            std::cerr << "rocBLAS ERROR: cannot determine home directory for rocBLAS log file"
+                      << std::endl;
+            std::cerr << "rocBLAS ERROR: turn off logging or create a home directory" << std::endl;
+            exit(-1);
```

Comment:
Remove the exit() call; libraries should not kill the host process.  If we can't log, output an error msg like you have above and let the app move on.

Reply:
fixed: removed exit() call